haystack/document_stores/elasticsearch.py

```diff
@@ -1271,14 +1602,75 @@ def update_embeddings(
     def delete_all_documents(
         self,
         index: Optional[str] = None,
-        filters: Optional[Dict[str, List[str]]] = None,
+        filters: Optional[Dict[str, Any]] = None,
         headers: Optional[Dict[str, str]] = None,
     ):
         """
         Delete documents in an index. All documents are deleted if no filters are passed.
 
         :param index: Index name to delete the document from.
         :param filters: Optional filters to narrow down the documents to be deleted.
+                        Filters are defined as nested dictionaries. The keys of the dictionaries can be a logical
```

Comment:
This docstring is nice, but we are repeating it so many times in this file that the file becomes pretty long and harder to navigate. How about we keep it in the long form for query and query_by_embedding. For the others we can shorten it a bit (maybe just keep one exampl) and add a link to the docs for more examples / a guide?

Reply:
Done.